### PR TITLE
Reuse Launcher orchestration in EngineTestKit

### LINKED ISSUE #2109: Consider support for PostDiscoveryFilters in EngineTestKit
The `EngineTestKit` is able to execute an `EngineDiscoveryRequest`.

This implies that a `PostDiscoveryFilter` is not respected since such filters are not applied by the `EngineTestKit`. Such filters are only applied by the `Launcher` when executing multiple test engines.

For example, `TagFilter.includeTags(...)` and `TagFilter.excludeTags(...)` are not supported by the `EngineTestKit`, and if they are registered via a `LauncherDiscoveryRequest` that is passed to the `EngineTestKit`, they are silently ignored.

This issue serves as a place to discuss the options for supporting `PostDiscoveryFilter` in EngineTestKit-based integration tests.

### PR DESCRIPTION
**Pull request by marcphilipp**
Resolves #2109.

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](https://junit.org/junit5/docs/snapshot/api/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [x] Change is documented in the [User Guide](https://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](https://junit.org/junit5/docs/snapshot/user-guide/#release-notes)
- [x] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

File: junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/EngineDiscoveryOrchestrator.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+	public LauncherDiscoveryResult discover(LauncherDiscoveryRequest request, String phase) {
+		Map<TestEngine, TestDescriptor> testEngineDescriptors = new LinkedHashMap<>();
+
+		for (TestEngine testEngine : this.testEngines) {
+			// @formatter:off
+            boolean engineIsExcluded = request.getEngineFilters().stream()
+                    .map(engineFilter -> engineFilter.apply(testEngine))
+                    .anyMatch(FilterResult::excluded);
+            // @formatter:on
```
Comment (jbduncan) [inline junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/EngineDiscoveryOrchestrator.java:61]: @marcphilipp Oops, looks like the code was not formatted properly.

Comment (marcphilipp) [reply junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/EngineDiscoveryOrchestrator.java:61]: Thanks, I hate it when IDEA does that. 😬

Comment (sormuras) [reply junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/EngineDiscoveryOrchestrator.java:61]: I hate `// @formatter:on/off` commented out directives too. 😁

Comment (marcphilipp) [reply junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/EngineDiscoveryOrchestrator.java:61]: Me too. :)

Comment (sormuras) [reply junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/EngineDiscoveryOrchestrator.java:61]: When is 2021?!
